tools/ProductConstructionService.ReproTool/Operations/FlatFlowTestOperation.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Maestro.Data;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.ProductConstructionService.Client;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Tools.Common;
+using GitHubClient = Octokit.GitHubClient;
+
+namespace ProductConstructionService.ReproTool.Operations;
+
+internal class FlatFlowTestOperation(
+    VmrDependencyResolver vmrDependencyResolver,
+    ILogger<FlatFlowTestOperation> logger,
+    GitHubClient ghClient,
+    BuildAssetRegistryContext context,
+    DarcProcessManager darcProcessManager,
+    IBarApiClient prodBarClient,
+    [FromKeyedServices("local")] IProductConstructionServiceApi localPcsApi) : Operation(logger, ghClient, context, localPcsApi)
+{
+    internal override async Task RunAsync()
+    {
+        await darcProcessManager.InitializeAsync();
+
+        var vmrRepos = await vmrDependencyResolver.GetVmrRepositoriesAsync(
+            "https://github.com/dotnet/dotnet",
+            "https://github.com/dotnet/sdk",
+            "main");
```

Comment:
This should probably come from the `Options.VmrBranch`?

Reply:
Hmm if we go with that, then the source mappings should already have the maestro-auth-test fork URI, you're right I think

Reply:
actually this will only work in the backflow